File: src/main/java/io/bloco/faker/components/Book.java
package io.bloco.faker.components;

import io.bloco.faker.FakerComponent;
import io.bloco.faker.FakerData;

public class Book extends FakerComponent {

    public Book(FakerData data) {
        super(data);
    }

    public String title() {
        return fetch("book.title");
    }

    public String author() {
        return parse("book.title");
    }

    public String publisher() {
        return fetch("book.publisher");
    }

    public String genre() {
        return fetch("book.genre");
    }
}

File: src/main/kotlin/io/bloco/faker/components/Book.kt
package io.bloco.faker.components

import io.bloco.faker.FakerComponent
import io.bloco.faker.FakerData

class Book(data: FakerData) : FakerComponent(data) {
    fun title(): String {
        return fetch("book.title")
    }

    fun author(): String {
        return parse("book.title")
    }

    fun publisher(): String {
        return fetch("book.publisher")
    }

    fun genre(): String {
        return fetch("book.genre")
    }
}
